app/presenters/outcome_presenter.rb

```diff
@@ -1,7 +1,25 @@
 require 'erubis'
 
 class OutcomePresenter < NodePresenter
-  class OutcomeTemplateMissing < StandardError; end
+  class ViewContext
+    def initialize(state)
+      @state = state
+      state_methods = state.methods - SmartAnswer::State.new(nil).methods
+      @state_reader_methods = state_methods.reject { |m| m =~ /=$/ }
+    end
+
+    def method_missing(method, *args, &block)
+      if @state_reader_methods.include?(method)
+        @state.send(method, *args, &block)
+      else
+        super
+      end
+    end
+
+    def get_binding
```

Comment:
I went with `#get_binding` as that's what's used in a number of examples in the Ruby docs. See the [`Kernel#binding` docs](http://ruby-doc.org/core-2.2.2/Kernel.html#method-i-binding), for example.

We can't name it `binding` as that results in an infinite loop of `#binding` trying to call itself.